tests/Fixer/Whitespace/BlankLineBeforeControlStatementFixerTest.php

```diff
@@ -0,0 +1,1046 @@
+     * @param string|null $input
+     */
+    public function testFixWithElseIf($expected, $input = null)
+    {
+        $this->fixer->configure(array(
+            'elseif',
+        ));
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithElseIf()
+    {
+        return array(
+            array(
+'<?php
+if ($c) {
+    echo $b;
+} 
+
+elseif ($b) {
+    echo $c;
+}',
+            ),
+            array(
+'<?php
+if ($c) {
+    echo $b;
+}
+
+elseif ($b) {
+    echo $c;
+}',
+'<?php
+if ($c) {
+    echo $b;
+} elseif ($b) {
+    echo $c;
+}',
+            ),
+            array(
+'<?php
+if ($c) 
+    echo $b;
+    
+elseif ($b)
+    echo $c;',
+            ),
+            array(
+'<?php
+if ($c) 
+    echo $b;
+
+elseif ($b) 
+    echo $c;',
+'<?php
+if ($c) 
+    echo $b;
+elseif ($b) 
+    echo $c;',
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerFixWithIf
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithIf($expected, $input = null)
+    {
+        $this->fixer->configure(array(
+            'if',
+        ));
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithIf()
+    {
+        return array(
+            array(
+'<?php
+if (true) {
+    echo $bar;
+}',
+            ),
+            array(
+'<?php
+$foo = $bar;
+
+if (true) {
+    echo $bar;
+}',
+'<?php
+$foo = $bar;
+if (true) {
+    echo $bar;
+}',
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerFixWithInclude
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithInclude($expected, $input = null)
+    {
+        $this->fixer->configure(array(
+            'include',
+        ));
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithInclude()
+    {
+        return array(
+            array(
+'<?php
+include "foo.php";',
+            ),
+            array(
+'<?php
+$foo = $bar;
+
+include "foo.php";',
+'<?php
+$foo = $bar;
+include "foo.php";',
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerFixWithIncludeOnce
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithIncludeOnce($expected, $input = null)
+    {
+        $this->fixer->configure(array(
+            'include_once',
+        ));
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithIncludeOnce()
+    {
+        return array(
+            array(
+'<?php
+include_once "foo.php";',
+            ),
+            array(
+'<?php
+$foo = $bar;
+
+include_once "foo.php";',
+'<?php
+$foo = $bar;
+include_once "foo.php";',
```

Comment:
hmm.. I don't think people have a lot of these types of blocks, therefor I wonder if it worth the additional dev. time and processing time to detect such cases.
If you want to go for it, please do so, but for me it would not be a show stopper for this PR